### PR TITLE
Changes to API validation for the Remove journey

### PR DESCRIPTION
**Pull request by mwestacott**
### JIRA link

(Vaguely) https://companieshouse.atlassian.net/browse/UAR-1281

### Change description

* DTO helper method changes to determine which journey is being followed
* Validation of the new remove 'is not proprietor of land' statement boolean field
* TODO comments added where work needs to be done for Remove (validation and filing)

This is a WIP - no tests and need to get a story created for this work

### Work checklist

- [ ] Tests added where applicable

### Merge instructions

We are committed to keeping commit history clean, consistent and linear. To achieve this, this commit should be structured as follows:

```
<type>[optional scope]: <description>
```

and contain the following structural elements:

- fix: a commit that patches a bug in your codebase (this correlates with PATCH in semantic versioning),
- feat: a commit that introduces a new feature to the codebase (this correlates with MINOR in semantic versioning),
- BREAKING CHANGE: a commit that has a footer `BREAKING CHANGE:` introduces a breaking API change (correlating with MAJOR in semantic versioning). A BREAKING CHANGE can be part of commits of any type,
- types other than `fix:` and `feat:` are allowed, for example `build:`, `chore:`, `ci:`, `docs:`, `style:`, `refactor:`, `perf:`, `test:`, and others,
- footers other than `BREAKING CHANGE: <description>` may be provided.

### CODE REVIEW

File: src/main/java/uk/gov/companieshouse/overseasentitiesapi/service/OverseasEntitiesService.java [hunk 1]
```diff
@@ -70,11 +70,15 @@ SubmissionType getSubmissionType(String requestId, String overseasEntityId) thro
...
+            ApiLogger.infoContext(requestId, String.format("Update submission with overseas entity number %s found",
                     entityNumber));
             return SubmissionType.UPDATE;
+        } else if (submissionOpt.get().isForRemove()) {
+            ApiLogger.infoContext(requestId, String.format("Remove submission with overseas entity number %s found",
+                    entityNumber));
+            return SubmissionType.REMOVE;
         } else {
-            ApiLogger.infoContext(requestId, "Submission without overseas entity number found");
+            ApiLogger.infoContext(requestId, "Registration submission without overseas entity number found");
```
Comment (markpit) [inline src/main/java/uk/gov/companieshouse/overseasentitiesapi/service/OverseasEntitiesService.java:81]: The Registration info message might be misleading, it makes it sound like some Registration submission can have an entity number.

Comment (mwestacott) [reply src/main/java/uk/gov/companieshouse/overseasentitiesapi/service/OverseasEntitiesService.java:81]: Yes, agree. It should probably just be "Registration submission found".

File: src/main/java/uk/gov/companieshouse/overseasentitiesapi/validation/RemoveValidator.java [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+import uk.gov.companieshouse.service.rest.err.Errors;
+
+import static uk.gov.companieshouse.overseasentitiesapi.validation.utils.UtilsValidators.setErrorMsgToLocation;
+import static uk.gov.companieshouse.overseasentitiesapi.validation.utils.ValidationUtils.getQualifiedFieldName;
+
+@Component
+public class RemoveValidator {
+
+    public Errors validate(RemoveDto removeDto, Errors errors, String loggingContext) {
+        if (removeDto.getIsNotProprietorOfLand() != null && removeDto.getIsNotProprietorOfLand() == false) {
```
Comment (mwestacott) [inline src/main/java/uk/gov/companieshouse/overseasentitiesapi/validation/RemoveValidator.java:17]: Thinking about this, logic should perhaps be:

`removeDto.getIsNotProprietorOfLand() == null || removeDto.getIsNotProprietorOfLand() == false`

Comment (HatiDev) [reply src/main/java/uk/gov/companieshouse/overseasentitiesapi/validation/RemoveValidator.java:17]: i agree

Comment (mwestacott) [reply src/main/java/uk/gov/companieshouse/overseasentitiesapi/validation/RemoveValidator.java:17]: Now not sure. The above holds true for _full_ validation within the API but perhaps not for _partial_ validation (where field may well be not present/null, at least early on in the journey). @HatiDev we will have to test this when you get the validator code running in Tilt.

**Pull request closed without merging**